### PR TITLE
Added pdf extension as well as doc|docx|xls to image_ext var

### LINKED ISSUE #3720: Preload links: Exclude PDF links from being automatically preloaded
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version ✅
- Used the search feature to ensure that the bug hasn’t been reported before ✅

**Describe the bug**
If you have links to PDF files into a post andyou move the mouse hover the links, it will automatically download the PDF files to the browser’s cache memory. This can a performance issue with big PDF files.

**To Reproduce**
1. Enable Preload Links option
2. Upload a PDF and set a link into a post
3. Open this post in a Chromium Browser
4. Move your mouse hover the PDF link
5. You’ll see in the Network tab of the console that the PDF is automatically downloaded. 

**Expected behavior**
PDF shouldn’t be downloaded automatically if you don’t click on it.

**Additional context**
This doesn’t occur when CDN is enable

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by jeawhanlee**
## Description

Added pdf|doc|docx|xls to image_ext variable to exlude from preload
Fixes #3720

## Type of change

- [x] Enhancement (non-breaking change which improves an existing functionality)

## Is the solution different from the one proposed during the grooming?

This solution is in no way different from the proposed one.

## How Has This Been Tested?

- Enable preload link in WPR
- Add pdf|doc|docx|xls file to media and add link to a page.
- Open browser network tab and hover over file link to see that link is not preloaded.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by remyperona — APPROVED** [review]

**Comment by Mai-Saad** [comment]
@jeawhanlee @Tabrisrp Fatal error exists after enabling WPR using the current branch (not sure if updating the branch to 3.11.0.4 will fix that or not). Can you please check it?
```
[14-Apr-2022 20:39:46 UTC] PHP Fatal error:  Uncaught Error: Call to undefined function WP_Rocket\Engine\Common\Queue\as_has_scheduled_action() in /var/www/new.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Engine/Common/Queue/AbstractASQueue.php:63
Stack trace:
#0 /var/www/new.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Engine/Common/Queue/AbstractASQueue.php(48): WP_Rocket\Engine\Common\Queue\AbstractASQueue->is_scheduled('rocket_rucss_pe...', Array, 'rocket-rucss')
#1 /var/www/new.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Engine/Optimization/RUCSS/Controller/Queue.php(57): WP_Rocket\Engine\Common\Queue\AbstractASQueue->schedule_recurring(1649968786, 60, 'rocket_rucss_pe...')
#2 /var/www/new.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Engine/Optimization/RUCSS/Admin/Subscriber.php(188): WP_Rocket\Engine\Optimization\RUCSS\Controller\Queue->schedule_pending_jobs_cron(60)
#3 /var/www/new.rocketlabsqa.ovh/htdocs/wp-includes/class-wp-hook.php(307): WP_Rocket\Engine\Optimization\RUCSS\Admin\Subscriber->sch in /var/www/new.rocketlabsqa.ovh/htdocs/wp-content/plugins/wp-rocket/inc/Engine/Common/Queue/AbstractASQueue.php on line 63
```

**Comment by jeawhanlee** [comment]
@Mai-Saad What branch produced this fatal error?

**Comment by Mai-Saad** [comment]
@jeawhanlee  `enhancement/3720-exclude-pdf-links-from-being-automatically-preloaded
`

**Comment by jeawhanlee** [comment]
Thanks @Mai-Saad 
Having a look at it...

**Comment by Mai-Saad** [comment]
@Tabrisrp @jeawhanlee we are preloading `.xlsx` and `.php` on hover, shouldn't it be excluded as well?
test page: https://new.rocketlabsqa.ovh/pdf-test/
Screenshot from 2022-04-15 11-27-57

**Comment by jeawhanlee** [comment]
@Mai-Saad Yes it should. Thanks

**Comment by jeawhanlee** [comment]
> @Tabrisrp @jeawhanlee we are preloading `.xlsx` and `.php` on hover, shouldn't it be excluded as well? test page: https://new.rocketlabsqa.ovh/pdf-test/ Screenshot from 2022-04-15 11-27-57

@Mai-Saad Here Fixed. please check again

File: inc/Engine/Preload/Links/Subscriber.php [hunk 1]
```diff
@@ -122,13 +122,13 @@ public function add_preload_script() {
...
 		$use_trailing_slash = $this->use_trailing_slash();
-		$images_ext         = 'jpg|jpeg|gif|png|tiff|bmp|webp|avif';
+		$images_ext         = 'jpg|jpeg|gif|png|tiff|bmp|webp|avif|pdf|doc|docx|xls|xlsx|php';
 
 		$config = [
 			'excludeUris'       => $this->get_uris_to_exclude( $use_trailing_slash ),
 			'usesTrailingSlash' => $use_trailing_slash,
 			'imageExt'          => $images_ext,
-			'fileExt'           => $images_ext . '|php|pdf|html|htm',
+			'fileExt'           => $images_ext . '|php|html|htm',
```
Comment (Mai-Saad) [inline inc/Engine/Preload/Links/Subscriber.php]: Thanks for the PR @jeawhanlee. I think `php` here is duplicated. what do you think?

Comment (jeawhanlee) [reply inc/Engine/Preload/Links/Subscriber.php]: @Mai-Saad You are right!

**Review by Mai-Saad — COMMENTED** [review]
Working as expected now. @piotrbak do we need to exclude any other extensions i.e `.json`?

**Comment by piotrbak** [comment]
@Mai-Saad I think what we have here is enough so far.

**Review by Mai-Saad — APPROVED** [review]

**Pull request merged**